src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+                                    incomingBitrate_))
+                    {
+                        updateEstimate = true;
+                    }
+                }
+            }
+            if (updateEstimate) {
+                // The first overuse should immediately trigger a new estimate.
+                // We also have to update the estimate immediately if we are
+                // overusing and the target bitrate is too high compared to
+                // what we are receiving.
+                RateControlInput input = new RateControlInput(detector
+                        .getState(), incomingBitrate.getRate(arrivalTimeMs),
+                        estimator.getVarNoise());
+
+                /**
+                 * @Todo remote.update returns nothing. This is fine since we
+                 * can retrieve the value from remoteRate.getLatestEstimate
+                 */
+                remoteRate.update(input, nowMs);
+                targetBitrateBps = remoteRate.getLatestEstimate();
+                updateEstimate = remoteRate.isValidEstimate();
+                ssrcs = Keys(ssrcs_);
+            }
+        }
+        if (updateEstimate) {
+            lastUpdateMs = nowMs;
+            /**
+             * @Todo Fix Problem with transforming Long to Integer
+             */
+            observer_.onReceiveBitrateChanged(null, targetBitrateBps);
+        }
+    }
+
+    void Process() {}
+
+    long timeUntilNextProcess()
+    {
+        long kDisabledModuleTime = 1000;
+        return kDisabledModuleTime;
+    }
+
+    void timeoutStreams(long nowMs)
+    {
+        synchronized (critSect) {
+            Iterator<Map.Entry<Long, Long>> itr = ssrcs_.entrySet().iterator();
+            while (itr.hasNext()) {
+                Long Key = (Long) itr.next().getValue();
+                if ((nowMs - (Long) ssrcs_.get(Key) > kStreamTimeOutMs)) {
+
+                    ssrcs_.remove((Long) Key);
+                }
+            }
+            if (ssrcs_.isEmpty()) {
+                // We can't update the estimate if we don't have any active streams.
+                interArrival = new InterArrival((kTimestampGroupLengthMs
+                        << kInterArrivalShift) / 1000,
+                        kTimestampToMs, true);
+                estimator = new OveruseEstimator(new OverUseDetectorOptions());
+                // We deliberately don't reset the first_packet_time_ms_
+                // here for now since we only probe for bandwidth in the
+                // beginning of a call right now.
+            }
+        }
+    }
+
+    void OnRttUpdate(long avg_rtt_ms,
+                     long max_rtt_ms)
+    {
+        synchronized (critSect) {
+            remoteRate.setRtt(avg_rtt_ms);
+        }
+    }
+
+    void removeStream(long ssrc)
+    {
+        synchronized (critSect)
+        {
+            ssrcs_.remove(ssrc);
+        }
+    }
+
+    boolean latestEstimate(List<Long> ssrcs,
+        long bitrateBps) {
```

Comment:
Okay, I didn't notice that. Thinking of this, I checked and saw this is supposed to be equivalent to 
`getLatestEstimate` in `RemoteBitrateEstimator`. In jitsi implementation, it returns `long` and [webrtc - line 63](https://chromium.googlesource.com/external/webrtc/+/master/webrtc/modules/remote_bitrate_estimator/include/remote_bitrate_estimator.h) returns `bool`. Although it is left to us to decide how to implement and use it. Thinking of the usage, I am going to make it return long. if true or valid, it `returns bitrateBps` else if false, it `returns -1`. It should be left for the calling method to check the value and use accordingly. let me know if this will be fine.